src/components/homePageMainContent/homePageMainContent.tsx

```diff
@@ -1,13 +1,32 @@
 import React from "react";
 import FeaturedContentComponent from "../featuredContent/featuredContent";
 import SwimLanes from "../swimlanes/swimLanes";
+import { useEffect, useState } from "react";
+import SwimLanesLoading from "../swimlanes/swimLanesLoading";
 
-const HomePageMainContent = ({ randomNumber, lanesWithNumItems }) => {
-  return (
+const HomePageMainContent = () => {
+  const [data, setData] = useState<any>({});
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const response = await fetch("/api/homepagedata");
+      const responseData = await response.json();
+      console.log(responseData);
+      setData(responseData);
+    };
+
+    console.log("fetching swim lane data");
+    fetchData();
+  }, []);
```

Comment:
In this case, this does have the effect of calling this once because there are no deps.

Reply:
this returns an empty array while the other does not.

Reply:
What returns an empty array? I meant that there are no deps in the array so the empty array, as the argument to the `useEffect`, makes it call the function once.